### PR TITLE
Example of provisioning dhcp range

### LINKED ISSUE #405: provisioningDHCPRange is missing from docs
I don't see it mentioned on https://openshift-kni.github.io/baremetal-deploy/4.4/Deployment.html#configuring-the-install-config-file_ipi-install-prerequisites

More info here:
https://github.com/openshift/installer/blob/master/docs/user/metal/customization_ipi.md

### PR DESCRIPTION
**Pull request by iranzo**
# Description

Add more examples to sample install-config.yaml

Preview at: https://deploy-preview-418--baremetal-deploy.netlify.app/4.4/deployment#additional-install-config-parameters_ipi-install-prerequisites

Fixes #405

## Type of change

Please select the appropiate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Versions:
- Hardware:

## Checklist

- [x] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not being merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

**Comment by rlopez133** [comment]
@iranzo we don't want to include the examples directly in the install-config.yaml . I think a better approach would be to place them in this table and describe them: https://deploy-preview-418--baremetal-deploy.netlify.app/4.4/deployment#additional-install-config-parameters_ipi-install-prerequisites

Similar to what you see in this table: https://github.com/openshift/installer/blob/master/docs/user/metal/customization_ipi.md

**Comment by iranzo** [comment]
> @iranzo we don't want to include the examples directly in the install-config.yaml . I think a better approach would be to place them in this table and describe them: https://deploy-preview-418--baremetal-deploy.netlify.app/4.4/deployment#additional-install-config-parameters_ipi-install-prerequisites
> 
> Similar to what you see in this table: https://github.com/openshift/installer/blob/master/docs/user/metal/customization_ipi.md

Check https://deploy-preview-418--baremetal-deploy.netlify.app/4.4/deployment#additional-install-config-parameters_ipi-install-prerequisites for the new 'optional parameters' table

**Comment by rlopez133** [comment]
> > @iranzo we don't want to include the examples directly in the install-config.yaml . I think a better approach would be to place them in this table and describe them: https://deploy-preview-418--baremetal-deploy.netlify.app/4.4/deployment#additional-install-config-parameters_ipi-install-prerequisites
> > Similar to what you see in this table: https://github.com/openshift/installer/blob/master/docs/user/metal/customization_ipi.md
> 
> Check https://deploy-preview-418--baremetal-deploy.netlify.app/4.4/deployment#additional-install-config-parameters_ipi-install-prerequisites for the new 'optional parameters' table

This looks great! I likely want to go down the customization_ipi list and make sure we capture them all prior to merge but yes you captured what I was thinking accurately.

File: documentation/ipi-install/modules/ipi-install-additional-install-config-parameters.adoc [hunk 1]
```diff
@@ -29,6 +29,15 @@ default name resolve correctly.
...
 
+.Optional parameters
+
+|===
+|Parameters |Default |Description
+|`provisioningDHCPExternal`| |Defines if external DHCP will be used or the one configured by installer
+|`provisioningDHCPRange`| |   i.e. 172.22.0.10,172.22.0.100  Defines the IP range to use for hosts on the provisioning network.
+|`provisioningNetworkCIDR`| | <CIDR> # i.e. fd00:1101::1/64
+|===
+
```
Comment (rlopez133) [inline documentation/ipi-install/modules/ipi-install-additional-install-config-parameters.adoc:40]: While we are at it, lets go ahead and add additional optional parameters. I'm going to include them here so you can include in your PR. I went ahead and did some column clean up too for easier readability. I hope it renders properly :D 

```
[cols="1,1,3", options="header"] 
.Optional Parameters
|===
|Parameters
|Default
|Description

|`provisioningDHCPExternal`
| false
|Defines if external DHCP will be used or the one configured by installer

|`provisioningDHCPRange`
| `172.22.0.10,172.22.0.100`
|Defines the IP range to use for hosts on the provisioning network.

|`provisioningNetworkCIDR`
| `172.22.0.0/24`
| The CIDR for the network to use for provisioning.

| `clusterProvisioningIP`
| `172.22.0.3`
| The IP within the cluster where the provisioning services run.

| `bootstrapProvisioningIP`
| `172.22.0.2`
| The IP on the bootstrap VM where the provisioning services run while the 
control plane is being deployed.

| `externalBridge`
| `baremetal`
| The name of the bridge of the hypervisor attached to the external network.

| `provisioningBridge`
| `provisioning`
| The name of the bridge on the hypervisor attached to the provisioning network.

| `defaultMachinePlatform`
| 
| The default configuration used for machine pools without a platform 
configuration.

| `bootstrapOSImage`
| 
| A URL to override the default operating system image for the bootstrap node. 
The URL must contain a sha256 hash of the image. 
Example https://mirror.openshift.com/rhcos-<version>-qemu.qcow2.gz?sha256=abc...

| `clusterOSImage`
|
| A URL to override the default operating system for cluster nodes. 
The URL must include a sha256 hash of the image. 
Example https://mirror.openshift.com/images/rhcos-<version>-openstack.qcow2.gz?sha256=abc...

|===
```

Comment (DirectedSoul1) [reply documentation/ipi-install/modules/ipi-install-additional-install-config-parameters.adoc:40]: /lgtm

Comment (rlopez133) [reply documentation/ipi-install/modules/ipi-install-additional-install-config-parameters.adoc:40]: We will also need to filter this list based on release

4.4 and 4.5 have all of the above from the list.

Version 4.3 **does not** have:
```
ProvisioningNetworkCIDR
ProvisioningDHCPExternal
ProvisioningDHCPRange
```

I got the list from https://github.com/openshift/installer/blob/release-4.5/pkg/types/baremetal/defaults/platform.go , feel free to change release to get the details you need.

Comment (iranzo) [reply documentation/ipi-install/modules/ipi-install-additional-install-config-parameters.adoc]: Done, with conditional for > 4.3 to show the extra parameters

**Comment by rlopez133** [comment]
We need to look into IPv6 defaults for provisioningDHCPRange , provisioningNetworkCIDR , clusterProvisioningIP , bootstrapProvisioningIP . Also need to find out if provisioningNetworkCIDR is optional when using IPv6 or required.

File: documentation/ipi-install/modules/ipi-install-additional-install-config-parameters.adoc [hunk 2]
```diff
@@ -5,7 +5,7 @@
 [id="additional-install-config-parameters_{context}"]
 = Additional `install-config` parameters
 
-This topic describes the required parameters, the `hosts` parameter, 
+This topic describes the required parameters, the `hosts` parameter,
 and the `bmc` `address` parameter for the `install-config.yaml` file.
```
Comment (rlopez133) [inline documentation/ipi-install/modules/ipi-install-additional-install-config-parameters.adoc:9]: This section describes the required parameters and optional parameters that are part of the `install-config.yaml` file

**Comment by iranzo** [comment]
Handled via #451

**Pull request closed without merging**